.github/workflows/on_push.yaml

```diff
@@ -0,0 +1,53 @@
+name: Publish to edge
+
+# On push to a "special" branch, we:
+# * always publish to charmhub at latest/edge/branchname
+# * always run tests
+# where a "special" branch is one of main/master or track/**, as
+# by convention these branches are the source for a corresponding
+# charmhub edge channel.
+
+on:
+  workflow_call:
+
+jobs:
+  lib-check:
+    # This job is the precondition of all jobs that need the Charmhub token.
+    # By disabling this job on forked repositories, we can achieve 
+    # disabling the publishing to Charmhub action on forked repositories 
+    # while still running tests on push events
+    if: github.repository_owner == 'canonical'
+    name: Check libraries
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Check libs
+        uses: canonical/charming-actions/check-libraries@1.0.3
+        with:
+          credentials: ${{ secrets.CHARMHUB_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+  tests:
+    name: Run tests
+    uses: ./.github/workflows/test.yaml
+  integration-tests:
+    name: Run integration tests
+    uses: ./.github/workflows/integration_test.yaml
```

Comment:
This kind of limits the aplicability of this workflow where, for example, the integration tests have some unique requirements (such as for Jenkins) and we can't use the standardised tests. Could this be changed to get the user to define the pre-conditions so that this can be used in more cases?

Reply:
Can you point me to those specific cases?

Reply:
Ok cool, do we need a follow up story or do you expect to be able to do it in your current story? Marking as resolved for this PR